content/Gems You Want To Have In Your Rails API - Juan Diego Rosales.md

```diff
@@ -0,0 +1,44 @@
+
+*Juan Diego Rosales is a software developer on the Rootstrap team and writes about his experience working on different projects. You can also follow Juan on [LinkedIn](www.linkedin.com/in/juan-diego-rosales-02a05a144).*
+
+Are you new to Rails? Or maybe just looking for any cool gem to use in your new API? Here is a list of gems we have on our Rails API Base that we consider a must-have in every Rails API project.
+
+## Authentication
+If you are building a Rails API, then you will probably need token based authentication and that is when [Devise Token Auth](https://github.com/lynndylanhurley/devise_token_auth) comes in. This gem is built on top of Devise, one of the most popular authentication gems for Rails which you may have already worked with. If that is the case, then you'll get familiar with Devise Token Auth very easily.
+
+With very little setup, this gem provides registration, sign in, sign out and reset password flows out of the box, as well as very useful helper functions. Also, it handles email confirmation if enabled and supports the use of multiple user models, giving you the possibility to manage the authentication of different types of users at the same time using groups.
+
+In comparison to other alternatives available, one of the best perks that comes with Devise Token Auth, which is inherited from Devise, is the very little effort you have to invest to get things up and running. On top of that, you can also customize the different flows mentioned if you want to offer a different experience, and while this may be a little more challenging, it's not rocket science!
+
+## Resource administration
+Most APIs typically need some level of resource administration and for that I would definitely recommend [Active Admin](https://github.com/activeadmin/activeadmin). Despite the fact it can have a steep learning curve, specially at the beginning, due to it's DSL (domain-specific language), once you get used to it you can have admin pages with CRUD actions for any resource in no time.
+
+Something very important to mention is that Active Admin has a large community and good documentation relative to other administration gems, such as Rails Admin or Administrate. Also, another strong feature is that you can configure the index filters according to your needs and they can be customized to use more that one attribute since they support [Ransack](https://github.com/activerecord-hackery/ransack). 
+
+If you need a custom page, you can have Active Admin render your own partial and you will probably need to make changes in the controller actions. This is not an easy task, but none of the other gems offer anything better.
+
+For some people, one of the downsides could be the design, since Active Admin does not come with a powerful CSS framework out of the box. Luckily, there are many complementary gems that provide a better looking theme, for example [Arctic Admin](https://github.com/cprodhomme/arctic_admin).
```

Comment:
what about referencing all the options besides giving a specific example: https://github.com/activeadmin/activeadmin/wiki/Themes ?